docs/develop/testing-cloe.rst

```diff
@@ -0,0 +1,180 @@
+a sucessful build usually includes sucessful test execution, such as when we
+run::
+
+    make package
+
+or::
+
+    make all
+
+It's therefore usually not necessary to explicitely run the unit tests, unless
+there is a failing test.
+
+To understand what's going on, it's recommended to check out the targets as
+defined in the ``Makefile.package`` file in the repository root.
+
+When tests are executed automatically, CMake collates available tests and runs
+them with its own command line output. If a test fails, all you see is that
+a test failed, but you won't see *why* it failed, nor will you directly be able
+to debug the test. In order to that, you need to find the test executable that
+CMake executed and run that yourself.
+
+For example, if you compiled with ``make all``, then you can check the
+``build/bin/`` directory for a test executable. This you can run directly to
+see a more detailed error report. You can also run the executable with GDB
+if you want to debug the test step-for-step::
+
+    gdb build/bin/test-engine
+
+When using GDB, it may be helpful to tell GDB to ``catch throw``
+(see `GDB Catchpoints<https://ftp.gnu.org/old-gnu/Manuals/gdb/html_node/gdb_30.html>`__).
+
+Integration Tests
+-----------------
+
+Running an integration test involves combining multiple Conan packages into
+a cohesive unit and running a Cloe simulation with a stack file. To streamline
+this we generally use ``cloe-launch`` to wrap the temporary installation of
+a virtual runtime environment, activating this environment, and running
+commands within the environment.
+
+Integration tests are located throughout the repository in ``tests/``
+directories. For example, the minimator plugin has this in the ``tests``
+directory, as of this writing::
+
+    plugins/minimator/tests
+    ├── config_minimator_infinite.json
+    ├── config_minimator_multi_agent_infinite.json
+    ├── config_minimator_smoketest.json
+    ├── profile_default.py
+    ├── test_minimator.bats
+    ├── test_minimator_multi_agent_smoketest.json
+    └── test_minimator_smoketest.json
+
+Each directory has different files, following the patterns below:
+
+``profile_*.py``
+    Each of these files are cloe-launch profiles (i.e., Conanfiles) which should
+    be used to test all of the stack files in the same directory.
+
+    Note that this may involve re-compilation of one or more parts of Cloe.
+    For example, one profile may use a configuration that has the server in
+    cloe-engine disabled, another may force a specifc boost version, and yet
+    another may use a different compiler or toolchain.
+
+    These can actually be used by Conan directly::
+
+        # mkdir virtualenv && cd virtualenv
+        # conan install ../profile_default.py
+        # source cloe_launch_env.sh
+        # source activate.sh
+        # source activate_run.sh
+
+    Using cloe-launch automates that process and adds some flags for a one-stop
```

Comment:
Maybe provide the exact `cloe-launch` command that does the job?

Reply:
done